File: samples.cs/SqlSet.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data.Common;
using System.IO;
using System.Linq;
using System.Text;
using DbExtensions;
using Samples.CSharp.Northwind;

namespace Samples.CSharp {
   
   public class SqlSetSamples {

      readonly DbConnection connection;
      readonly SqlSet<Product> productSet;

      public SqlSetSamples(string connectionString, TextWriter log) {
         
         this.connection = DbFactory.CreateConnection(connectionString);
         this.productSet = new SqlSet<Product>(this.connection, new SqlBuilder("SELECT * FROM Products"), log);
      }

      public object AreThereAnyProducts() {
         return productSet.Any();
      }

      public object DoAllProductsHaveUnitPrice() {
         return productSet.All("NOT UnitPrice IS NULL");
      }

      public object DoSomeProductsAreOutOfStock() {
         return productSet.Any("UnitsInStock = 0");
      }

      public object HowManyProductsAreOutOfStock() {
         return productSet.Count("UnitsInStock = 0");
      }

      public Product FirstProduct() {
         return productSet.First();
      }

      public Product SecondProduct() {
         return productSet.Skip(1).First();
      }

      public Product FirstOutOfStockProduct() {
         return productSet.First("UnitsInStock = 0");
      }

      public IEnumerable Top5ProductsWithLowestStock() {
         return productSet.Where("UnitsInStock > 0")
            .OrderBy("UnitsInStock")
            .Take(5)
            .Select(r => new { Name = r.GetString(0), UnitsInStock = r.GetInt16(1) }, "ProductName, UnitsInStock")
            .AsEnumerable();
      }

      public object NamesOfOutOfStockProducts() {
         
         return productSet.Where("UnitsInStock = 0")
            .Select(r => r.GetString(0), "ProductName")
            .ToArray();
      }

      public Product GetSpecificProduct() {
         return productSet.SingleOrDefault("ProductID = 5");
      }
   }
}

File: samples.vb/SqlSet.vb
Imports System
Imports System.Collections
Imports System.Collections.Generic
Imports System.Data.Common
Imports System.IO
Imports System.Linq
Imports System.Text
Imports DbExtensions
Imports Samples.VisualBasic.Northwind

Public Class SqlSetSamples

   ReadOnly connection As DbConnection
   ReadOnly productSet As SqlSet(Of Product)

   Public Sub New(ByVal connectionString As String, ByVal log As TextWriter)
      Me.connection = DbFactory.CreateConnection(connectionString)
      Me.productSet = Me.connection.Set(Of Product)(New SqlBuilder("SELECT * FROM Products"), log)
   End Sub

   Public Function AreThereAnyProducts() As Object
      Return productSet.Any()
   End Function

   Public Function DoAllProductsHaveUnitPrice() As Object
      Return productSet.All("NOT UnitPrice IS NULL")
   End Function

   Public Function DoSomeProductsAreOutOfStock() As Object
      Return productSet.Any("UnitsInStock = 0")
   End Function

   Public Function HowManyProductsAreOutOfStock() As Object
      Return productSet.Count("UnitsInStock = 0")
   End Function

   Public Function FirstProduct() As Product
      Return productSet.First()
   End Function

   Public Function SecondProduct() As Product
      Return productSet.Skip(1).First()
   End Function

   Public Function FirstOutOfStockProduct() As Product
      Return productSet.First("UnitsInStock = 0")
   End Function

   Public Function Top5ProductsWithLowestStock() As IEnumerable

      Return productSet.Where("UnitsInStock > 0") _
         .OrderBy("UnitsInStock") _
         .Take(5) _
         .Select(Function(r) New With {.Name = r.GetString("ProductName"), .UnitsInStock = r.GetInt16("UnitsInStock")}) _
         .AsEnumerable()

   End Function

   Public Function NamesOfOutOfStockProducts() As Object

      Return productSet.Where("UnitsInStock = 0") _
         .Select(Function(r) r.GetString(0), "ProductName") _
         .ToArray()

   End Function

   Public Function GetSpecificProduct() As Product
      Return productSet.SingleOrDefault("ProductID = 5")
   End Function

End Class
